graphql_client/src/reqwest.rs

```diff
@@ -0,0 +1,15 @@
+//! A concrete client implementation over HTTP with reqwest.
+
+use crate::GraphQLQuery;
+
+/// Use the provided reqwest::Client to post a GraphQL request.
+pub async fn post_graphql<Q: GraphQLQuery, U: reqwest::IntoUrl>(
+    client: &reqwest::Client,
+    url: U,
+    variables: Q::Variables,
+) -> Result<crate::Response<Q::ResponseData>, reqwest::Error> {
+    let body = Q::build_query(variables);
+    let reqwest_response = client.post(url).json(&body).send().await?;
+
+    Ok(reqwest_response.json().await?)
+}
```

Comment:
This may be too inflexible. Let's wait for feedback.

Reply:
I'd say so. It's missing error handling at the HTTP level at least. See [this code](https://gitlab.kitware.com/utils/rust-ghostflow/-/blob/master/ghostflow-github/src/client.rs#L310) for how GitHub's GQL API is handled.

Reply:
So basically return the whole response as error in case status != 200? I haven't worked with GraphQL APIs in a long time, but if I remember correctly, APIs weren't consistent on that point. I'll check the graphql spec tomorrow to see if it says anything about it, but I think it's HTTP agnostic.

Reply:
I couldn't find anything HTTP-specific in the spec, but there is [something on graphql.org](https://graphql.org/learn/serving-over-http/#response). The status code isn't important, the response should always be a GraphQL response body. I'm still going to try to return more information, and fail in a useful way if the response body isn't a valid GraphQL response.

Reply:
Looks like the `reqwest::Error` might already give enough context.

Reply:
(https://docs.rs/reqwest/0.11.4/reqwest/struct.Error.html)